### DISCUSSION TITLE
Actions not triggering on PR

### DISCUSSION THREAD

**Discussion by nerdoza** [community/community#26395] [Actions]
<p>The last few days I’ve been experiencing intermittent issues with tests not running on a pull-request.  I haven’t seen this occurring on the main branch, but I haven’t made many commits to it so I can’t say for certain that it’s not happening there.</p>
<p>Here’s an example from the logs where the test for the first 3 pushes (which were all pushed together) were skipped, but the next push 4 hours later the test ran for, as well as the one after it.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/4/e4c144712145b666ef245e8437e429d10dab584c.png" data-download-href="/uploads/short-url/wDEW4uz182AqT0V1KhKPqo67YxS.png?dl=1" title="Capture" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2280384/181097463-56984f84-a7fe-425a-ab17-86e1c7e1aabd.png" alt="Capture" data-base62-sha1="wDEW4uz182AqT0V1KhKPqo67YxS" width="690" height="337" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/4/e4c144712145b666ef245e8437e429d10dab584c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture</span><span class="informations">721×353 28 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here’s the workflow file if it matters:</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
    types:
      - opened
      - reopened
      - synchronize

name: Master

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    timeout-minutes: 10
    steps:
      - name: Checkout Repository
        uses: actions/checkout@v2

      - name: Use Node.js v12
        uses: actions/setup-node@v1.4.0
        with:
          node-version: 12

      - name: Setup MySQL
        uses: mirromutth/mysql-action@v1.1
        with:
          mysql version: '5.7'
          mysql database: 'test'
          mysql root password: 'password'

      - name: Install
        run: npm ci

      - name: Set ENV
        run: echo "MYSQL_PASS=\"password\"" &gt; .env

      - name: Test
        run: npm test
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/bayssmekanique">@bayssmekanique</a>,</p>
<p>Here are few points you may need to know, please check with them in your case:</p>
<ol>
<li>
<p>When you push multiple commits at once, the workflows will be triggered only for the latest commit contained in the this push. It is designed as that.</p>
</li>
<li>
<p>In addition, when you use the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow" rel="noopener nofollow ugc"><strong>GITHUB_TOKEN</strong></a> to authenticate when pushing commits, the push event will not trigger a new workflow run.</p>
<blockquote>
<p>When you use the repository’s <strong>GITHUB_TOKEN</strong> to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
</li>
</ol>

**Comment by nerdoza** [comment]
<p><a class="mention" href="/brightran">@brightran</a>,  I understand the first point, but as you can see from the timestamps, the action didn’t run on the last commit of the 3 I pushed together.</p>
<p>As for the second point, all of these pushes were made using the exact same “git push” command from the same location with the same authentication mechanism, so I don’t think that would have any effect here.</p>
<p>Any other thoughts/suggestions?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/bayssmekanique">@bayssmekanique</a>,</p>
<p>Looks like, there indeed are some issues occur. Maybe the GitHub services have some issues occur recently.<br>
I noticed a GitHub Incident below that may be relevant:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://dka575ofm4ao0.cloudfront.net/pages-favicon_logos/original/36420/akacZEQQfOBdc7ftyxJt" class="site-icon" width="32" height="32">
      <a href="https://www.githubstatus.com/incidents/4mzhxxpwgvqg" target="_blank" rel="noopener nofollow ugc">githubstatus.com</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/46307996/181097465-549c8b47-4f21-453d-a2e9-90609454cc93.png" class="thumbnail onebox-avatar" width="120" height="120">

<h3><a href="https://www.githubstatus.com/incidents/4mzhxxpwgvqg" target="_blank" rel="noopener nofollow ugc">Incident on 2020-11-02 12:14 UTC</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>This incident had been resolved few days ago.</p>
